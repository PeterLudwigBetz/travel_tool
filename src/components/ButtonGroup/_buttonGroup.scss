.button-group{
    width: auto;
    display: flex;
    height: 36px;
    align-items: center;
}

.bg-btn {
    height: 36px;
    width: auto;
    border: 1px solid #D4D4D4;
    border-radius: 1px;
    background-color: #F8F8F8;
    color: #999999;
    font-family: "DIN Pro";
    font-size: 14px;
    font-weight: 500;
    line-height: 14px;
    margin-right: 10px;
    padding: 9px 21px;

  &.bg-btn--active {
        border: 1px solid #3359DB;
        border-radius: 1px;
        background-color: #3359DB;
        color: #FFFFFF;
    }

  &.bg-btn--with-badge {
        display: flex;
        justify-content: space-around;
        align-items: center;

      span.bg-btn__label {
            color: #999999;
            font-family: "DIN Pro";
            font-size: 14px;
            font-weight: 500;
            line-height: 14px;
            text-align: right;
        }

      span.badge{
            height: 15px;
            width: 15px;
            background-color: #3359DB;
            border-radius: 50%;
            color: #FFFFFF;
            font-family: "DIN Pro";
            font-size: 12px;
            font-weight: 500;
            line-height: 15px;
            text-align: center;
            margin-left: 4px;
        }

    }

}


