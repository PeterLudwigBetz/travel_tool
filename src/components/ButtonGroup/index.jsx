import React, { PureComponent } from 'react';
import './buttonGroup.scss';

class ButtonGroup extends PureComponent {
  render() {
    return (
      <div className="button-group">
        <button type="button" className="bg-btn bg-btn--active">
              All
        </button>

        <button type="button" className="bg-btn bg-btn--with-badge">
          <span className="label">
            Open Reqeusts
          </span>
          <span className="badge">
            3
          </span>
        </button>

        <button type="button" className="bg-btn">
            Past Requests
        </button>
      </div>
    );
  }
}


export default ButtonGroup;
