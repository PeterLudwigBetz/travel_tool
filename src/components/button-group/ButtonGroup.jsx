import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import Button from '../buttons/Buttons';
import './buttonGroup.scss';

class ButtonGroup extends PureComponent {
  defaultState = {
    allActive: false,
    openActive: false,
    pastActive: false,
  }

  state = {
    ...this.defaultState,
    allActive: true,
  };

  fetchRequests = (query, active) => {
    const { fetchUserRequests, limit } = this.props;
    const queryString = `/?page=1&limit=${limit}`;
    this.setState(active);
    fetchUserRequests(`${queryString}${query}`);
  }

  renderApprovalsButton () {
    return (
      <Fragment>
        <Button
          buttonClass="bg-btn bg-btn--active"
          text="All"
        />
        <Button
          buttonClass="bg-btn bg-btn--with-badge"
          text="Pending Approvals"
          responsiveText="Open"
          badge={3}
          showBadge
          badgeClass="bg-btn--with-badge__approvals--inactive"
        />
        <Button
          buttonClass="bg-btn"
          responsiveText="Past"
          text="Past Approvals"
        />
      </Fragment>
    );
  }

  renderRequestsButton (openRequestsCount, pastRequestsCount) {
    const { pastActive, openActive, allActive } = this.state;
    return (
      <Fragment>
        <Button
          buttonClass={`bg-btn ${allActive ? 'bg-btn--active' : ''}`}
          text="All"
          buttonId="all-button"
          onClick={() => this.fetchRequests('', {...this.defaultState, allActive: true })}
        />
        <Button
          buttonClass={`bg-btn bg-btn--with-badge ${ openActive ? 'bg-btn--active' : ''}`}
          text="Open Requests"
          responsiveText="Open"
          buttonId="open-button"
          disabled={openRequestsCount === 0}
          badge={openRequestsCount}
          showBadge={openRequestsCount > 0}
          badgeClass={openActive ? 'bg-btn--with-badge--active' : 'bg-btn--with-badge--inactive'}
          onClick={() => this.fetchRequests('&status=open', {...this.defaultState, openActive: true })}
        />
        <Button
          buttonClass={`bg-btn ${pastActive ? 'bg-btn--active' : ''}`}
          disabled={pastRequestsCount === 0}
          text="Past Requests"
          responsiveText="Past"
          buttonId="past-button"
          onClick={() => this.fetchRequests('&status=past', {...this.defaultState, pastActive: true})}
        />
      </Fragment>
    );
  }

  render() {
    const { buttonsType, openRequestsCount, pastRequestsCount } = this.props;
    return (
      <div className="button-group">
        { buttonsType === 'approvals' && this.renderApprovalsButton()}
        { buttonsType === 'requests' &&
          this.renderRequestsButton(openRequestsCount, pastRequestsCount) }
      </div>
    );
  }
}

ButtonGroup.propTypes = {
  openRequestsCount: PropTypes.number,
  pastRequestsCount: PropTypes.number,
  fetchUserRequests: PropTypes.func,
  limit: PropTypes.number,
  buttonsType: PropTypes.string.isRequired,
};

ButtonGroup.defaultProps = {
  limit: 10,
  openRequestsCount: null,
  pastRequestsCount: null,
  fetchUserRequests: null,
};


export default ButtonGroup;
