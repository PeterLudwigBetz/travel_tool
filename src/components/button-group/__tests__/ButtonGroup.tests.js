import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import ButtonGroup from '../ButtonGroup';


const props = {
  openRequestsCount: 1,
  pastRequestsCount: 1,
  limit: 10,
  fetchUserRequests: sinon.spy(() => Promise.resolve()),
  buttonsType: 'requests'
};

const defaultState = {
  allActive: false,
  openActive: false,
  pastActive: false,
}

describe('<ButtonGroup />', () => {
  it('should render successfully', () => {
    const wrapper = shallow(<ButtonGroup {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls the fetchRequests method when the `all` button is clicked', () => {
    const wrapper = mount(<ButtonGroup {...props} />);
    const spy = sinon.spy(wrapper.instance(), 'fetchRequests');
    const allButton = wrapper.find('#all-button');
    const newState = {
      ...defaultState,
      allActive: true
    };
    allButton.simulate('click');
    expect(spy.calledOnce).toEqual(true);
    expect(spy.calledWith('', newState));
    expect(wrapper.state()).toEqual(newState)
  });

  it('calls the fetchRequests method when the `open requests` button is clicked', () => {
    const wrapper = mount(<ButtonGroup {...props} />);
    const spy = sinon.spy(wrapper.instance(), 'fetchRequests');
    const openButton = wrapper.find('#open-button');
    const newState = {
      ...defaultState,
      openActive: true
    };
    openButton.simulate('click');
    expect(spy.calledOnce).toEqual(true);
    expect(spy.calledWith('', newState));
    expect(wrapper.state()).toEqual(newState);
  });

  it('calls the fetchRequests method when the `past requests` button is clicked', () => {
    const wrapper = mount(<ButtonGroup {...props} />);
    const spy = sinon.spy(wrapper.instance(), 'fetchRequests');
    const pastButton = wrapper.find('#past-button');
    const newState = {
      ...defaultState,
      pastActive: true
    };
    pastButton.simulate('click');
    expect(spy.calledOnce).toEqual(true);
    expect(spy.calledWith('', newState));
    expect(wrapper.state()).toEqual(newState);
  });
})
