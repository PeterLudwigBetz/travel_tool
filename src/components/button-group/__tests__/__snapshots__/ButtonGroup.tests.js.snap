// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroup /> should render successfully 1`] = `
<div
  className="button-group"
>
  <Button
    altText=""
    badge={null}
    badgeClass=""
    buttonClass="bg-btn bg-btn--active"
    buttonId="all-button"
    disabled={false}
    imageClass="mdl-Icon"
    imageSrc=""
    onClick={[Function]}
    responsiveText=""
    showBadge={false}
    text="All"
    textClass=""
  />
  <Button
    altText=""
    badge={1}
    badgeClass="bg-btn--with-badge--inactive"
    buttonClass="bg-btn bg-btn--with-badge "
    buttonId="open-button"
    disabled={false}
    imageClass="mdl-Icon"
    imageSrc=""
    onClick={[Function]}
    responsiveText="Open"
    showBadge={true}
    text="Open Requests"
    textClass=""
  />
  <Button
    altText=""
    badge={null}
    badgeClass=""
    buttonClass="bg-btn "
    buttonId="past-button"
    disabled={false}
    imageClass="mdl-Icon"
    imageSrc=""
    onClick={[Function]}
    responsiveText="Past"
    showBadge={false}
    text="Past Requests"
    textClass=""
  />
</div>
`;
